### DISCUSSION TITLE
Mise goes to github on hook-env

### DISCUSSION THREAD

**Discussion by olfway** [jdx/mise#3861] [Troubleshooting and bug reports]
I have a local proxy broken and now on every cd I have 5s timeout on "mise hook-env"

I tried to run it manually and I get:
```
~ ❯ cd Downloads/
mise ERROR [/Users/olfway/.config/mise/config.toml] ubi:ameshkov/dnslookup@1.11.0: error sending request for url (https://api.github.com/repos/ameshkov/dnslookup/releases): operation timed out
mise ERROR error resolving versions
```

my config:
```
❯ cat /Users/olfway/.config/mise/config.toml
[tools]
"ubi:ameshkov/dnslookup" = "1.11.0"
```

Previously I had version "latest" but even when I write "1.11.0" it still goes to github

I was assumed that it shouldn't go to github.com even with "latest" after a tool was installed

**Comment by syhol** [comment]
Try `export MISE_TRACE=1` to produce some logs. You can disable it with `export MISE_TRACE=`.

But I don't see why it would try to call github on cd. Do you have any hooks configured?

**Reply by olfway** [reply]
I've checked my configs and found that mise cache was empty, you can reproduce it like this:
```
mkdir "$(pwd)/mise-test"
echo 'kubectl latest' > "$(pwd)/mise-test/.tool-versions"
   
env -i HOME="$(pwd)/mise-test" \
MISE_TIMINGS=1 \
http_proxy=http://1.2.3.4:1234 \
https_proxy=http://1.2.3.4:1234 \
/opt/homebrew/bin/mise --cd "$(pwd)/mise-test" hook-env

[TIME] 1 mise::cli::run hook-env 15.02s
mise ERROR [~/.tool-versions] kubectl@latest: error sending request for url (https://mise-versions.jdx.dev/aqua-registry/kubernetes/kubectl/registry.yaml): operation timed out
mise ERROR error resolving versions
mise ERROR Run with --verbose or MISE_VERBOSE=1 for more information
```

So, with only one configured tool, each shell prompt takes 15s
I'm not sure how it's best to handle it, but it's almost impossible to use console in this case

Once mise downloaded cache - it working fast even if I didn't installed tools

**Reply by jdx** [reply]
I think we should change the aqua logic to reuse stale registry info unless it's actually installing something, or maybe if it's like a month old.
Mise shouldn't be checking for versions at all on hook-env too so I think there's a couple of things to fix here

**Comment by Marukome0743** [comment]
Hi @olfway — this should now be resolved in recent versions of mise.

The original report covered both an exact version (`1.11.0`) and `latest` being resolved during `hook-env`, causing network requests and long shell-prompt delays when the cache was empty or the network was unavailable.

The relevant fixes are:

- #7976 skips remote version fetching for exact, uninstalled versions when running in prefer-offline mode.
- #8500 extends the same behavior to `latest`.
- #9627 prevents remote-version cache refreshes from being forced in prefer-offline paths.

`hook-env` runs in prefer-offline mode, so it should now avoid contacting GitHub or the versions host for these requests. Remote validation is deferred until an operation that actually needs it, such as `mise install`.

Could you try the original reproduction with a recent mise release? If it still makes a network request, please share the output of:

```sh
mise --version
MISE_TRACE=1 mise hook-env
```

Thanks for the clear reproduction!

*This comment was generated by an AI coding assistant.*
